mergedSchema.js

```diff
@@ -0,0 +1,46 @@
+import { mergeSchemas, introspectSchema, makeRemoteExecutableSchema } from "graphql-tools"
+import { createHttpLink } from "apollo-link-http"
+import fetch from "node-fetch"
+
+import localSchema from "./schema"
+
+export default async function mergedSchema() {
+  const convectionLink = createHttpLink({
+    fetch,
+    uri: process.env.CONVECTION_GRAPH_URL,
+    headers: {
+      Authorization: `Bearer ${process.env.CONVECTION_TOKEN}`,
+    },
+  })
+
+  const convectionSchema = await makeRemoteExecutableSchema({
+    schema: await introspectSchema(convectionLink),
+    link: convectionLink,
+  })
+
+  const linkTypeDefs = `
+    extend type Submission {
+      artist: Artist
+    }
+  `
+
+  return mergeSchemas({
+    schemas: [localSchema, convectionSchema, linkTypeDefs],
+    // Prefer others over the local MP schema.
+    onTypeConflict: (_leftType, rightType) => {
+      // console.warn(`[!] Type collision ${rightType}`)
+      return rightType
+    },
+    resolvers: mergeInfo => ({
+      Submission: {
+        artist: {
+          fragment: `fragment SubmissionArtist on Submission { artist_id }`,
+          resolve: (parent, args, context, info) => {
+            const id = parent.artist_id
+            return mergeInfo.delegate("query", "artist", { id }, context, info)
```

Comment:
This is basically at runtime generating a query like:

```graphql
{
  artist(id: "<result of artist_id>") {
    # ...
  }
}
```

Reply:
Interesting note is that we may not want to have root fields for each type that we stitch into other types. Maybe we can use the `node` root field for this? 🤔

Reply:
The `mergeSchemas` author explained to me that this should be done in a few steps:

1. Add the required root field to the original schema.
2. Merge the schemas and setup the delegator to use the root field.
3. Transform the merged schema to remove the root field (the delegator refers to the original schema).